pkg/api/api.go

```diff
@@ -613,11 +614,14 @@ func (hs *HTTPServer) registerRoutes() {
 	r.Get("/avatar/:hash", requestmeta.SetSLOGroup(requestmeta.SLOGroupHighSlow), hs.AvatarCacheServer.Handler)
 
 	// Snapshots
-	r.Post("/api/snapshots/", reqSnapshotPublicModeOrSignedIn, hs.getCreatedSnapshotHandler())
 	r.Get("/api/snapshot/shared-options/", reqSignedIn, hs.GetSharingOptions)
+
+	r.Post("/api/snapshots/", reqSnapshotPublicModeOrCreate, hs.getCreatedSnapshotHandler())
 	r.Get("/api/snapshots/:key", routing.Wrap(hs.GetDashboardSnapshot))
```

Comment:
There seem to be a bunch of access options for snapshots, which makes me a bit confused:
* public snapshot mode (in which case we don't do any access checks);
* snapshot key which allows accessing a snapshot (so don't need auth checks?);
* the rest, which is protected by RBAC middleware

Did I get it right?

Reply:
* The problem here is we use the same endpoints for public mode and regular mode. Therefore, we need to check if it is public OR it has permissions.
* Snapshot key access is the endpoint to retrieve and render a snapshot. It should be public.
* The rest CRUD endpoints (create, delete and list snapshots) are using RBAC middleware